File: Convert_Integer_to_the_Sum_of_Two_No-zero_Integers.java
class Solution {

    public boolean hasZero(int x) {
        while (x > 0) {
            if (x % 10 == 0) return true; 
            x /= 10;    
            
        }
        return false;
    }
    public int[] getNoZeroIntegers(int n) {
        for (int i = 1; i < n; i++) {
            int find = n - i;
            if (!hasZero(i) && !hasZero(find)) {
                return new int[] {i, find};
            }
        }
        return new int[]{};
    }
}

File: Convert_Integer_to_the_Sum_of_Two_No-zero_Integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        def has_zero(x):
            return '0' in str(x)
        for i in range(n):
            find = n - i
            if not has_zero(find) and not has_zero(i):
                return [i, find]